experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricData.ts

```diff
@@ -57,6 +57,7 @@ export type MetricData = SingularMetricData | HistogramMetricData;
 export interface InstrumentationLibraryMetrics {
   instrumentationLibrary: InstrumentationLibrary;
   metrics: MetricData[];
+  errors: unknown[];
```

Comment:
Interesting. :thinking: I think I've not seen any other language SDKs exposing these errors to the exporter. Is there a spec requirement for this? :thinking:

Reply:
Yeah, https://github.com/open-telemetry/opentelemetry-specification/pull/2495

Reply:
Ah, I see. Thank you! :slightly_smiling_face: 
Do you know of backends that support adding the errors to the exported metrics? 

I am wondering if logging them in the `MetricReader` itself would be an option. :thinking: Otherwise each exporter would have to log or ignore them, which would cause users to have a different experience with Async Instruments depending on which Exporter they are using. This complicate troubleshooting for them.

Reply:
The way I read the spec the errors don't get reported to the exporter, but to the "caller" of `collect` which would be the MetricReader right?

Reply:
Yes. For push exporters, only the `MetricReader` is the caller. 
However, our pull exporters extend `MetricReader`, so pull exporters are callers too.

Reply:
This is only because of the choice to model the exporter as a MetricReader which was optional. If an exporter is going to be a MetricReader then it should have all the responsibilities of metric readers including error handling if that is one of them.

Reply:
We can also move these errors out of `InstrumentationLibraryMetrics` or `ResourceMetrics`, to be a structure parallel with `ResourceMetrics`, while keeping the exporter interface unchanged so exporters don't have to handle these errors. But that would require additional effort to restructure the return value type of `MetricReader.collect`. Like: 

```ts
interface CollectionResult {
  resourceMetrics: ResourceMetrics; 
  errors: unknown[];
}
```

Reply:
I will leave that decision to you since you've done the primary work on metrics, just let us know what you decide.

Reply:
> We can also move these errors out of `InstrumentationLibraryMetrics` or `ResourceMetrics`, to be a structure parallel with `ResourceMetrics`, while keeping the exporter interface unchanged so exporters don't have to handle these errors. But that would require additional effort to restructure the return value type of `MetricReader.collect`.

I like this solution. :slightly_smiling_face:

Reply:
Updated, PTAL again :)

Reply:
Looks great! :slightly_smiling_face: Thanks for taking the time to address this :slightly_smiling_face: